tensorrt_llm/_torch/models/checkpoints/base_checkpoint_loader.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+from torch import nn
+
+from tensorrt_llm._torch.model_config import ModelConfig
+from tensorrt_llm._torch.models.checkpoints.auto_mapper import \
+    AutoCheckpointMapper
+from tensorrt_llm._torch.models.checkpoints.base_config_loader import \
+    BaseConfigLoader
+from tensorrt_llm._torch.models.checkpoints.base_weight_loader import \
+    BaseWeightLoader
+from tensorrt_llm._torch.models.checkpoints.base_weight_mapper import \
+    BaseWeightMapper
+from tensorrt_llm._torch.models.modeling_utils import \
+    CHECKPOINT_LOADER_FORMAT_DEFAULT_MAPPING
+
+
+class BaseCheckpointLoader(ABC):
+
+    @abstractmethod
+    def get_default_weight_loader(self) -> BaseWeightLoader:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_default_config_loader(self) -> BaseConfigLoader:
+        raise NotImplementedError
+
+    @abstractmethod
+    def cleanup(self) -> None:
+        raise NotImplementedError
+
+    @property
```

Comment:
Do we really need to have these properties? Can't we just have raw fields and cut down on boilerplate?